contracts/RandomBeaconHistory.cdc

```diff
@@ -36,25 +49,172 @@ access(all) contract RandomBeaconHistory {
         ///
         /// @param randomSourceHistory The random source to record
         ///
+        /// The Flow protocol makes sure to call this function once per block as a system call. The transaction
+        /// comes at the end of each block so that the current block's entry becomes available only in the child
+        /// block.
+        ///
         access(all) fun heartbeat(randomSourceHistory: [UInt8]) {
 
+            assert(
+                // random source must be at least 128 bits
+                randomSourceHistory.length >= 128 / 8,
+                message: "Random source must be at least 128 bits"
+            )
+
             let currentBlockHeight = getCurrentBlock().height
+            // init lowestBlockHeight if it is not set yet
             if RandomBeaconHistory.lowestHeight == nil {
                 RandomBeaconHistory.lowestHeight = currentBlockHeight
             }
+            // Create & save Backfiller if it is not saved yet
+            if RandomBeaconHistory.borrowBackfiller() == nil {
+                RandomBeaconHistory.account.save(<-create Backfiller(), to: /storage/randomBeaconHistoryBackfiller)
+            }
+            let backfiller = RandomBeaconHistory.borrowBackfiller() ?? panic("Problem borrowing backfiller")
```

Comment:
```suggestion
            var maybeBackfiller = RandomBeaconHistory.borrowBackfiller()
            if maybeBackfiller == nil {
                RandomBeaconHistory.account.save(<-create Backfiller(), to: /storage/randomBeaconHistoryBackfiller)
                maybeBackfiller = RandomBeaconHistory.borrowBackfiller()
            }
            let backfiller = maybeBackfiller ?? panic("Problem borrowing backfiller")
```

This should be les operations which is really important for the system transaction.
(disclaimer: untested code)

Reply:
good point: https://github.com/onflow/flow-core-contracts/pull/416/commits/a14b5e23f1e1902e5a14a88fce8f5cf322d4a290